### PR TITLE
mvc-thymeleaf extension

### PR DESCRIPTION
**Pull request by ddovnar**
New extension module for working with Thymeleaf mvc templates.

**How to use**

**1. Add class and properties into ResourceConfig:**

```java
...
config.property(ThymeleafMvcFeature.TEMPLATE_BASE_PATH, "WEB-INF/templates/");
config.register(ThymeleafMvcFeature.class);
...
```

**2. Create template files and localized messages files**

Templates files should be placed in the resources folder in the project. For example:

jersey-thymeleaf-resources

`index.html`

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<p th:text="${firstName}">DEFAULT</p>
<p th:text="${lastName}">DEFAULT</p>

<p th:text="#{label.title}">List</p>
</body>
</html>
```
Messages properties files for localization:
`index.properties`

```
label.title = List of Products
```
`index_de.properties`
```
label.title = Produktliste
```
**3. Example how to use in the controller class**

```java
@Path("/demo1")
    @GET
    public Viewable mvc() {
        return new Viewable("/index.html", "Fruit Index Page");
    }

    @GET
    @Template(name = "/index.html")
    @Path("/demo2")
    @Produces(MediaType.TEXT_HTML)
    public Map<String, Object> mvc1(@QueryParam("lang") String lang) {
        Map<String, Object> model = new HashMap<>();
        model.put("firstName", "Demo");
        model.put("lastName", "User");
        model.put("lang", lang);
        return model;
    }
```

### CODE REVIEW

**Comment by jansupol** [comment]
Please see our [wiki](https://github.com/eclipse-ee4j/jersey/wiki/Missing-ECA,-Missing-Sign-off-by). Also, each file must have a copyright header with the copyright year (2024 this year) and the license, just like the other files in the repository.

**Comment by jansupol** [comment]
@ddovnar Thank you for adding the copyright header, but the copyright year needs to be 2024. We also have the issue with ECA, as the check says your email dmitriy.dovnar@ar*on... cannot be found in Eclipse.

File: ext/mvc-thymeleaf/pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jansupol) [inline ext/mvc-thymeleaf/pom.xml:1]: [INFO] --- glassfish-copyright:2.4:check (default-cli) @ jersey-mvc-thymeleaf ---
/home/jenkins/agent/workspace/ci-cd-integration_PR-5616/ext/mvc-thymeleaf/pom.xml: No copyright

Comment (ddovnar) [reply ext/mvc-thymeleaf/pom.xml:1]: resolved

**Review by jansupol — APPROVED** [review]

**Review by senivam — APPROVED** [review]

**Pull request merged**